codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -50,29 +50,31 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         this.connection = checkNotNull(connection, "connection");
 
         // Add a flow state for the connection.
-        connection.connectionStream().setProperty(FlowState.class,
-                new FlowState(connection.connectionStream(), initialWindowSize));
+        connection.connectionStream().remoteFlowState(
+                new DefaultFlowState(new ReducedFlowState(connection.connectionStream()), initialWindowSize));
 
         // Register for notification of new streams.
         connection.addListener(new Http2ConnectionAdapter() {
-            @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.setProperty(FlowState.class, new FlowState(stream, 0));
+                // TODO: (scottmitch): we used to have a create and open...but now they are separate.
```

Comment:
@nmittler @buchgr - The events for which `ReducedFlowState` is being utilized are not complete (hense the `TODO` statements) but the idea is streams can be in states where flow controlled frames cannot be exchanged and thus we don't require the full amount of state to account for these streams.  The two scenarios this applies for are:
1. Streams are created and inserted into the priority tree if a PRIORITY frame is the only frame to be exchanged on the stream (received or sent).
2. Streams can be closed and remain the priority tree because dependents of it are still "active".

I'm thinking if we can get the trigger points for this the change may be worth it because we don't know how long streams will stay in these states, and we might as well either wait to allocate the memory (for scenario 1) or allow GC to reclaim the majority of the state as soon as possible (for scenario 2).

Trigger points for each respective event:
1. Stream is created, but will not immediately transition out of idle.
2. Stream is "removed" but the `prioritizableForTree() != 0`.

I think 2 can be detected relatively easily.  However this TODO is in reference to not being able to detect 1 without interface changes (hense the awkward trigger points).  For example if `onStreamAdded(..)` would indicate the state the stream was going to then this would provide this trigger point.  However this may require changes to `createStream(...)` and the distinction between `onStreamAdded(..)` and `onStreamActive(..)` becomes less clear.  This is what I wanted to have a discussion about...is this worth it...is there another way? :)